bundles/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/internal/PlugwiseBinding.java

```diff
@@ -68,142 +74,170 @@ protected void removeBindingProvider(PlugwiseBindingProvider bindingProvider) {
 
-                    // the config-key enumeration contains additional keys that we
-                    // don't want to process here ...
-                    if ("service.pid".equals(key)) {
-                        continue;
-                    }
+                if (device != null) {
+                    stick.plugwiseDeviceCache.add(device);
+                }
+            }
 
-                    Matcher matcher = EXTRACT_PLUGWISE_CONFIG_PATTERN.matcher(key);
-                    if (!matcher.matches()) {
-                        logger.error("given plugwise-config-key '" + key
-                                + "' does not follow the expected pattern '<PlugwiseId>.<mac|port>'");
-                        continue;
-                    }
+        }
 
-                    matcher.reset();
-                    matcher.find();
+    }
 
-                    String plugwiseID = matcher.group(1);
+    private PlugwiseDevice createPlugwiseDevice(String deviceType, String MAC, String deviceName) {
+
+        PlugwiseDevice device = null;
+
+        if ("circleplus".equals(deviceType) || "circleplus".equals(deviceName)) {
+            // for backwards compatibility a device with the name 'circleplus' always creates a CirclePlus
+            device = new CirclePlus(MAC, stick, deviceName);
+            logger.info("Plugwise created Circle+ with name: {} and MAC address: {}", deviceName, MAC);
+        } else if ("circle".equals(deviceType) || deviceType == null) {
+            // for backwards compatibility a device without a deviceType always creates a Circle
+            device = new Circle(MAC, stick, deviceName);
+            logger.info("Plugwise created Circle with name: {} and MAC address: {}", deviceName, MAC);
+        } else if ("scan".equals(deviceType)) {
+            device = new Scan(MAC, stick, deviceName);
+            logger.info("Plugwise created Scan with name: {} and MAC address: {}", deviceName, MAC);
+        } else if ("sense".equals(deviceType)) {
+            device = new Sense(MAC, stick, deviceName);
+            logger.info("Plugwise created Sense with name: {} and MAC address: {}", deviceName, MAC);
+        } else if ("stealth".equals(deviceType)) {
+            device = new Stealth(MAC, stick, deviceName);
+            logger.info("Plugwise created Stealth with name: {} and MAC address: {}", deviceName, MAC);
+        } else if ("switch".equals(deviceType)) {
+            device = new Switch(MAC, stick, deviceName);
+            logger.info("Plugwise created Switch with name: {} and MAC address: {}", deviceName, MAC);
+        } else {
+            logger.error(
+                    "Plugwise can not create device with name: '{}' because it has an unknown device type: '{}'. "
+                            + "Known device types are: circle|circleplus|scan|sense|stealth|switch",
+                    deviceName, deviceType);
+        }
+        return device;
+    }
```

Comment:
`logger.info` -> `logger.debug` or `logger.trace`

`logger.error` -> `logger.warn` unless system stability is affected